sql/session.go

```diff
@@ -88,10 +88,11 @@ func (s *BaseSession) Address() string { return s.addr }
 func (s *BaseSession) Client() Client { return s.client }
 
 // Set implements the Session interface.
-func (s *BaseSession) Set(key string, typ Type, value interface{}) {
+func (s *BaseSession) Set(ctx context.Context, key string, typ Type, value interface{}) error {
```

Comment:
Should be a *sql.Context

Reply:
This would create a weird circular reference that I don't really like.